src/react/hooks/useFragment.ts

```diff
@@ -52,11 +52,17 @@ extends Omit<
 //   canonizeResults?: boolean;
 // }
 
-export interface UseFragmentResult<TData> {
-  data: TData | undefined;
-  complete: boolean;
-  missing?: MissingTree;
-}
+export type UseFragmentResult<TData> =
+  | {
+      data: TData;
+      complete: true;
+      missing?: MissingTree;
```

Comment:
I believe `missing` should be `never` here since a `complete` result would imply there are no missing fields.

```suggestion
      missing?: never;
```

Reply:
Good point!